db/post_migrate/20230601132637_backfill_status_preview_card.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class BackfillStatusPreviewCard < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  class PreviewCardStatusJoin < ApplicationRecord
+    self.table_name = 'preview_cards_statuses'
+    self.primary_key = 'status_id'
+  end
+
+  class Status < ApplicationRecord; end
+
+  def up
+    PreviewCardStatusJoin.in_batches do |preview_card_status_join|
+      Status.where(id: preview_card_status_join.ids)
+            .update_all( # rubocop:disable Rails/SkipsModelValidations
+              'preview_card_id = (SELECT preview_card_id FROM preview_cards_statuses WHERE status_id = statuses.id LIMIT 1)'
```

Comment:
Should this be ordered so its deterministic?

Reply:
`in_batches` will order on `status_id`

As for the inner query, I don't think it matters: I have seen no evidence for duplicates on `status_id` not sharing a `preview_card_id` as well, and if they did, I'm not sure we could identify a more “legitimate” one.